dove/src/index/resolver/git.rs

```diff
@@ -29,10 +29,22 @@ pub fn resolve(ctx: &Context, git: &Git) -> Result<PathBuf, Error> {
             repo_path = repo_path.join("._tmp_dove_checkout_dir_");
         }
 
-        if let Err(err) = checkout(checkout_params, &repo_path) {
-            fs::remove_dir_all(&repo_path)?;
-            return Err(err);
-        }
+        checkout(checkout_params, &repo_path).map_err(|err| {
```

Comment:
This code looks **much** more complicated than it was. 
Wouldn't it be better if we just add `if repo_path.exists()` around line number 33?